File: SeriesGuide/src/main/java/com/battlelancer/seriesguide/ui/QuickCheckInActivity.java
package com.battlelancer.seriesguide.ui;

import android.os.Bundle;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.NotificationManagerCompat;
import com.battlelancer.seriesguide.SgApp;
import com.battlelancer.seriesguide.enums.TraktAction;
import com.battlelancer.seriesguide.service.NotificationService;
import com.battlelancer.seriesguide.ui.dialogs.CheckInDialogFragment;
import com.battlelancer.seriesguide.ui.dialogs.GenericCheckInDialogFragment;
import com.battlelancer.seriesguide.util.TraktTask;
import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;

/**
 * Blank activity, just used to quickly check into a show/episode on GetGlue/trakt.
 */
public class QuickCheckInActivity extends FragmentActivity {

    @SuppressWarnings("FieldCanBeLocal") private CheckInDialogFragment checkInDialogFragment;

    public interface InitBundle {

        String EPISODE_TVDBID = "episode_tvdbid";
    }

    @Override
    protected void onCreate(Bundle arg0) {
        // make the activity show the wallpaper, nothing else
        setTheme(android.R.style.Theme_Holo_Wallpaper_NoTitleBar);
        super.onCreate(arg0);

        int episodeTvdbId = getIntent().getIntExtra(InitBundle.EPISODE_TVDBID, 0);
        if (episodeTvdbId == 0) {
            finish();
            return;
        }

        checkInDialogFragment = CheckInDialogFragment.newInstance(this, episodeTvdbId);
        if (checkInDialogFragment == null) {
            finish();
            return;
        }
        // show check-in dialog
        checkInDialogFragment.show(getSupportFragmentManager(), "checkin-dialog");
    }

    @Override
    protected void onStart() {
        super.onStart();

        EventBus.getDefault().register(this);
    }

    @Override
    protected void onStop() {
        super.onStop();

        EventBus.getDefault().unregister(this);
    }

    @SuppressWarnings("unused")
    @Subscribe
    public void onEvent(GenericCheckInDialogFragment.CheckInDialogDismissedEvent event) {
        // if check-in dialog is dismissed, finish ourselves as well
        finish();
    }

    @SuppressWarnings("unused")
    @Subscribe
    public void onEvent(TraktTask.TraktActionCompleteEvent event) {
        if (event.mTraktAction != TraktAction.CHECKIN_EPISODE) {
            return;
        }
        // display status toast about trakt action
        event.handle(this);

        // dismiss notification on successful check-in
        if (event.mWasSuccessful) {
            NotificationManagerCompat manager = NotificationManagerCompat.from(
                    getApplicationContext());
            manager.cancel(SgApp.NOTIFICATION_EPISODE_ID);
            // replicate delete intent
            NotificationService.handleDeleteIntent(this, getIntent());
        }
    }
}

File: app/src/main/java/com/battlelancer/seriesguide/traktapi/QuickCheckInActivity.kt
// SPDX-License-Identifier: Apache-2.0
// Copyright 2013-2018, 2021-2024 Uwe Trottmann

package com.battlelancer.seriesguide.traktapi

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.battlelancer.seriesguide.R
import com.battlelancer.seriesguide.notifications.NotificationService
import com.battlelancer.seriesguide.traktapi.GenericCheckInDialogFragment.CheckInDialogDismissedEvent
import com.battlelancer.seriesguide.traktapi.TraktTask.TraktActionCompleteEvent
import com.battlelancer.seriesguide.ui.BaseThemeActivity
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

/**
 * Blank activity, just used to quickly check into a show/episode.
 */
class QuickCheckInActivity : BaseThemeActivity() {

    override fun getCustomTheme(): Int {
        // make the activity show the wallpaper, nothing else
        return R.style.Theme_SeriesGuide_Wallpaper
    }

    override fun configureEdgeToEdge() {
        // Do nothing.
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val episodeId = intent.getLongExtra(EXTRA_LONG_EPISODE_ID, 0)
        if (episodeId == 0L) {
            finish()
            return
        }

        // show check-in dialog
        if (!CheckInDialogFragment.show(this, supportFragmentManager, episodeId)) {
            finish()
        }
    }

    override fun onStart() {
        super.onStart()
        EventBus.getDefault().register(this)
    }

    override fun onStop() {
        super.onStop()
        EventBus.getDefault().unregister(this)
    }

    @Subscribe
    fun onEvent(@Suppress("UNUSED_PARAMETER") event: CheckInDialogDismissedEvent?) {
        // if check-in dialog is dismissed, finish ourselves as well
        finish()
    }

    // Post on main thread so Toast works
    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onEvent(event: TraktActionCompleteEvent) {
        if (event.traktAction != TraktAction.CHECKIN_EPISODE) {
            return
        }
        // display status toast about trakt action
        event.handle(this)

        // dismiss notification on successful check-in
        if (event.wasSuccessful) {
            NotificationService.deleteNotification(applicationContext, intent)
        }
    }

    companion object {

        private const val EXTRA_LONG_EPISODE_ID = "episode_id"

        @JvmStatic
        fun intent(episodeId: Long, context: Context): Intent {
            return Intent(context, QuickCheckInActivity::class.java)
                .putExtra(EXTRA_LONG_EPISODE_ID, episodeId)
                .addFlags(
                    Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS
                )
        }
    }
}